rpc/src/v1/helpers/light_fetch.rs

```diff
@@ -332,8 +333,11 @@ impl LightFetch {
 			// match them with their numbers for easy sorting later.
 			let bit_combos = filter.bloom_possibilities();
 			let receipts_futures: Vec<_> = self.client.ancestry_iter(filter.to_block)
-				.take_while(|ref hdr| BlockId::Number(hdr.number()) != filter.from_block)
-				.take_while(|ref hdr| BlockId::Hash(hdr.hash()) != filter.from_block)
+				.take_while(|ref hdr|{
+					BlockId::Hash(hdr.hash()) != filter.from_block
+					&& BlockId::Number(hdr.number()) != filter.from_block
```

Comment:
this is a bug. this take_while will never return `false` therefore this iterator will never end

Reply:
`take_while(a).take_while(b)` is equivalent to `take_while(a && b)`, right? 
Should we check for the `BlockId::Latest` case as well?

Reply:
Seems correct to me (also when testing). I can still switch back to using two `take_while` (does not make a lot of difference as it combines correctly).

Reply:
`take_while(a).take_while(b)` is equal to `take_while(a || b)` and this is a big difference